sqldb/sqlc/queries/payments.sql

```diff
@@ -0,0 +1,174 @@
+    ha.first_hop_amount_msat,
+    ha.route_total_time_lock,
+    ha.route_total_amount,
+    ha.route_source_key,
+    hr.resolution_type,
+    hr.resolution_time,
+    hr.failure_source_index,
+    hr.htlc_fail_reason,
+    hr.failure_msg,
+    hr.settle_preimage
+FROM payment_htlc_attempts ha
+LEFT JOIN payment_htlc_attempt_resolutions hr ON hr.attempt_index = ha.attempt_index
+WHERE ha.payment_id = $1
+ORDER BY ha.attempt_time ASC;
+
+-- name: FetchAllInflightAttempts :many
+-- Fetch all inflight attempts across all payments
+SELECT
+    ha.id,
+    ha.attempt_index,
+    ha.payment_id,
+    ha.session_key,
+    ha.attempt_time,
+    ha.payment_hash,
+    ha.first_hop_amount_msat,
+    ha.route_total_time_lock,
+    ha.route_total_amount,
+    ha.route_source_key
+FROM payment_htlc_attempts ha
+WHERE NOT EXISTS (
+    SELECT 1 FROM payment_htlc_attempt_resolutions hr 
+    WHERE hr.attempt_index = ha.attempt_index
+);
+
+-- name: FetchHopsForAttempt :many
+SELECT
+    h.id,
+    h.htlc_attempt_index,
+    h.hop_index,
+    h.pub_key,
+    h.scid,
+    h.outgoing_time_lock,
+    h.amt_to_forward,
+    h.meta_data,
+    m.payment_addr AS mpp_payment_addr,
+    m.total_msat AS mpp_total_msat,
+    a.root_share AS amp_root_share,
+    a.set_id AS amp_set_id,
+    a.child_index AS amp_child_index,
+    b.encrypted_data,
+    b.blinding_point,
+    b.blinded_path_total_amt
+FROM payment_route_hops h
+LEFT JOIN payment_route_hop_mpp m ON m.hop_id = h.id
+LEFT JOIN payment_route_hop_amp a ON a.hop_id = h.id
+LEFT JOIN payment_route_hop_blinded b ON b.hop_id = h.id
+WHERE h.htlc_attempt_index = $1
+ORDER BY h.hop_index ASC;
+
+-- name: FetchHopsForAttempts :many
+SELECT
+    h.id,
+    h.htlc_attempt_index,
+    h.hop_index,
+    h.pub_key,
+    h.scid,
+    h.outgoing_time_lock,
+    h.amt_to_forward,
+    h.meta_data,
+    m.payment_addr AS mpp_payment_addr,
+    m.total_msat AS mpp_total_msat,
+    a.root_share AS amp_root_share,
+    a.set_id AS amp_set_id,
+    a.child_index AS amp_child_index,
+    b.encrypted_data,
+    b.blinding_point,
+    b.blinded_path_total_amt
+FROM payment_route_hops h
+LEFT JOIN payment_route_hop_mpp m ON m.hop_id = h.id
+LEFT JOIN payment_route_hop_amp a ON a.hop_id = h.id
+LEFT JOIN payment_route_hop_blinded b ON b.hop_id = h.id
+WHERE h.htlc_attempt_index IN (sqlc.slice('htlc_attempt_indices')/*SLICE:htlc_attempt_indices*/);
+
+-- name: FetchPaymentLevelFirstHopCustomRecords :many
+SELECT
+    l.id,
+    l.payment_id,
+    l.key,
+    l.value
+FROM payment_first_hop_custom_records l
+WHERE l.payment_id = $1;
+
+-- name: FetchRouteLevelFirstHopCustomRecords :many
+SELECT
+    l.id,
+    l.htlc_attempt_index,
+    l.key,
+    l.value
+FROM payment_attempt_first_hop_custom_records l
+WHERE l.htlc_attempt_index IN (sqlc.slice('htlc_attempt_indices')/*SLICE:htlc_attempt_indices*/);
+
+-- name: FetchHopLevelCustomRecords :many
+SELECT
+    l.id,
+    l.hop_id,
+    l.key,
+    l.value
+FROM payment_hop_custom_records l
```

Comment:
just a reminder that when we use these, we must always wrap them with the sqdb.ExecuteBatch helper so that we never overflow